addons/pos_restaurant/static/src/overrides/models/pos_store.js

```diff
@@ -249,9 +249,30 @@ patch(PosStore.prototype, {
         }
         return super.getDefaultSearchDetails();
     },
+    async removeDoneOrders(table) {
+        // Orders with integer ids can be found in the server.
+        const syncedOrderIds = table["<-pos.order.table_id"]
+            .map((o) => o.id)
+            .filter((id) => typeof id === "number");
+
+        const ordersStatus =
+            syncedOrderIds.length > 0
+                ? await this.data.read("pos.order", syncedOrderIds, ["state"])
+                : [];
+
+        // cancelled, posted, paid and invoiced orders can be removed
```

Comment:
Paid orders can be retained, because if the user has already loaded them from the ticket_screen, they will no longer be loaded.

Otherwise LGTM

Reply:
True. We should take that into account.